pkgs/os-specific/linux/firmware/fwupd/default.nix

```diff
@@ -112,16 +115,18 @@ let
 
   self = stdenv.mkDerivation rec {
     pname = "fwupd";
-    version = "1.7.2";
+    version = "1.7.6";
 
     # libfwupd goes to lib
     # daemon, plug-ins and libfwupdplugin go to out
     # CLI programs go to out
     outputs = [ "out" "lib" "dev" "devdoc" "man" "installedTests" ];
 
-    src = fetchurl {
-      url = "https://people.freedesktop.org/~hughsient/releases/fwupd-${version}.tar.xz";
-      sha256 = "sha256-hjLfacO6/Fk4fNy1F8POMaWXoJAm5E9ZB9g4RnG5+DQ=";
+    src = fetchFromGitHub {
```

Comment:
> Why switch away from tarballl?

Github is the upstream VCS for fwupd (per https://fwupd.org/lvfs/docs/developers ). I think using the VCS repo rather than a tarball makes it easier for NixOS users/developers to find the upstream code.

More likely that bots will be able to propose PRs for new releases.

I'm happy to switch back to the tarball though.

> Also the hash is wrong.

Oops, sorry. While developing this PR, I had initially used https://github.com/fwupd/fwupd/commit/2cd58847fc456de0c6a2270e67fea2842f3c07cb as the `rev`, and set the hash to be the hash of that. When I eventually set `rev` to be a tag, I forgot to update the hash, and the Nix build cache was happy to use its cached version (rather than download the tag's copy, and alert me that the hash was different).

Fixed.

Reply:
Generally, we prefer using the official tarballs when available, since that is most likely what is tested by upstream.

Reply:
SGTM. Fixed.